settings/README.md

```diff
@@ -0,0 +1,238 @@
+
+The core token required for handling the settings is
+`ISettingRegistry` ([documentation](https://jupyterlab.github.io/jupyterlab/coreutils/modules/isettingregistry.html)). To use it,
+you first need to install its npm package:
+
+```bash
+jlpm add @jupyterlab/coreutils
+```
+
+Once this is done. You can import the interface in your code.
+
+```ts
+// src/index.ts#L6-L6
+
+import { ISettingRegistry } from '@jupyterlab/coreutils';
+```
+
+To see how we can access the settings, let's have a look at
+`src/index.ts`.
+
+```ts
+// src/index.ts#L13-L17
+
+const extension: JupyterFrontEndPlugin<void> = {
+  id: PLUGIN_ID,
+  autoStart: true,
+  requires: [ISettingRegistry],
+  activate: (app: JupyterFrontEnd, settings: ISettingRegistry) => {
+```
+
+The `ISettingRegistry` is passed to the `activate` function as an
+argument (variable `settings`) in addition to the JupyterLab application
+(variable `app`). You request that dependency with the property
+`requires: [ISettingRegistry],`. It lists the additional arguments
+you want to inject into the `activate` function in the `JupyterFontEndPlugin`.
+
+But before going further, you need to define the settings of your
+extension. This is done through a [JSON Schema](https://json-schema.org/understanding-json-schema/).
+In the example it is called `schema/plugin.json`.
+
+<!-- prettier-ignore-start -->
+```json5
+// schema/plugin.json
+
+{
+  "title": "My settings",
+  "description": "Settings of the settings example.",
+  "type": "object",
+  "properties": {
+    "limit": {
+      "type": "integer",
+      "title": "Limit",
+      "description": "This is an example of an integer setting.",
+      "default": 25
+    },
+    "flag": {
+      "type": "boolean",
+      "title": "Simple flag",
+      "description": "This is an example of a boolean setting.",
+      "default": false
+    }
+  }
+}
+
+```
+<!-- prettier-ignore-end -->
+
+The _title_ is the entry shown in the JupyterLab Advanced Settings. The
+_description_ entry is a more detail explanation of the extension using
+those settings. The _type_ is a mandatory key require by [JSON Schema](https://json-schema.org/understanding-json-schema/reference/type.html).
+For all extensions, this will be an `object` as the settings are defined
+by a _dictionary_. Then the most important entry is `properties` describing a mapping of _setting id_ and the associated properties.
+
+The naming of the file needs to follow a strict convention. When using
+settings, your extension name must be structured as _package name_**:**_settings name_. The settings file must be named _settings name_. In the example, the package name is:
+
+<!-- prettier-ignore-start -->
+```json5
+// package.json#L2-L2
+
+"name": "@jupyterlab-examples/settings",
+```
+<!-- prettier-ignore-end -->
+
+and the extension id is:
+
+```ts
+// src/index.ts#L8-L8
+
+const PLUGIN_ID = '@jupyterlab-examples/settings:plugin';
+```
+
+Therefore the settings file must be named `plugin.json`.
+
```

Comment:
Maybe we could a less canonical name for the file, such as `my-settings-example.json`, and have `PLUGIN_ID = '@jupyterlab-examples/settings:my-settings-example';`